src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/LocalizationModelDeserializer.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Localization;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects
+{
+    internal static class LocalizationModelDeserializer
+    {
+        private const string PostActionIndexPrefix = "postActions[";
+        private const string PostActionIndexSuffix = "]";
+
+        private const string ManualInstructionIndexPrefix = "manualInstructions[";
+        private const string ManualInstructionIndexSuffix = "]";
+
+        public static ILocalizationModel Deserialize(JObject data)
+        {
+            var locModel = new LocalizationModel();
+            var parameterLocalizations = new Dictionary<string, ParameterSymbolLocalizationModel>();
+
+            // Property names are in format: symbols.framework.choices.0.description
+            // Split them using '.' and store together with the localized string (property value).
+            IEnumerable<(IEnumerable<string> nameParts, string localizedString)> stringsWithNames = data.Properties()
+                .Where(p => p.Value.Type == JTokenType.String)
+                .Select(p => (p.Name.Split('.').AsEnumerable(), p.Value.ToString()))
+                .ToList();
```

Comment:
I'm not crazy in love with all this Linq and .Split, I would prefer some more .StartsWith("symbols") and .IndexOf('.') stuff, but I guess its fine as long as it doesn't show up on perf...

Reply:
I'm not talking just about this statement, but all others in this method and called methods like `LoadSymbolModels`, `LoadPostActionModels`...

Reply:
addition to comment below:
probably we can do here:
`data.Properties().Where(p => p.Value.Type == JTokenType.String).Select(p => (p.Name, p.Value.ToString()))` and pass it to respective private methods, which evaluate which strings they should act upon.